shpc/main/modules/__init__.py

```diff
@@ -409,43 +396,38 @@ def write_version_file(self, uri, tag, default_choice=False):
 
         # First or last installed
         elif self.settings.default_version == "last_installed":
-            self._default_version_last_installed(version_file, tag)
+            self._set_default_version(version_file, tag)
 
         elif self.settings.default_version == "first_installed":
-            self._default_version_first_installed(version_file, tag, default_choice)
+            first_installed = True if len(os.listdir(version_dir)) == 1 else False
+            if first_installed:
+                self._set_default_version(version_file, tag)
 
     def update_version_file(self, version_dir):
         """
         Given a module directory, update the version file.
         """
-        # The module directory has the version (which is deleted)
-        version_file = os.path.join(version_dir, ".version")
-
-        # If a version file doesn't exist, nothing to update
-        if not os.path.exists(version_file):
-            return
-
+        # Count how many versions we actually have
         found = [x for x in os.listdir(version_dir) if x != ".version"]
 
         # No versions left, remove the whole directory
         if len(found) == 0:
             self._cleanup(version_dir)
             return
 
-        uri = os.path.basename(version_dir)
+        # With the following settings, .version doesn't contain an actual version
+        # so there is nothing to update
+        if self.settings.default_version in [False, None, True, "sys_module"]:
+            return
 
-        # Only one result, present we just installed it
+        # Here, there is at least one version installed, and default_version
+        # is either first_installed or last_installed
         if len(found) == 1:
-            return self.write_version_file(uri, found[0])
-
-        # Use filesystem dates to derive a tag for the last installed
-        dates = [[utils.creation_date(os.path.join(version_dir, x)), x] for x in found]
-
-        # Sort based on wanting last installed (later time) or first (earliest)
-        reverse = False if self.settings.default_version == "first_installed" else True
-        # reverse True, latest -> earliest (return latest) otherwise return earliest
-        dates = sorted(dates, key=itemgetter(0), reverse=reverse)
-        tag = dates[0][1]
+            tag = found[0]
```

Comment:
This needs to be sorted depending on the date.

Reply:
`found[0]` is used when `len(found) == 1`, see above. That's the same shortcut as there used to be line 439

Reply:
Let's revisit this after the changes above.